### PR TITLE
Use require.resolve to determine the path of cross-env

### PR DESCRIPTION
**Pull request by mbleigh**

### CODE REVIEW

**Comment by mbleigh** [comment]
@tinaliang can you test this one out on the Windows laptop as well?

File: lib/deploy/lifecycleHooks.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ var logger = require('../logger');
 var path = require('path');
 
 function runCommand(command, childOptions) {
-  var translatedCommand = path.resolve(__dirname, '..', '..', 'node_modules', '.bin', 'cross-env') + ' ' + command;
+  var translatedCommand = path.resolve(require.resolve('cross-env'), '..', 'bin', 'cross-env.js') + ' ' + command;
```
Comment (laurenzlong) [inline lib/deploy/lifecycleHooks.js]: Is the addition of .js intentional?

**Comment by mbleigh** [comment]
Yes. Directly invoking from the package dir instead of the symlinked bin

**Comment by tinaliang** [comment]
I'm receiving an invalid character/Microsoft JScript compilation error.

**Comment by mbleigh** [comment]
@tinaliang can you update and try again? I think I've solved the bug you ran into.

**Comment by tinaliang** [comment]
@mbleigh The command is expanding out to `C:\Program Files\nodejs\node.exe C:\Users\User\firebase-tools\node_modules\cross-env\dist\bin\cross-env.js ...` and throwing an error because of the space in "C:\Program Files". It's parsing that as "C:\Program".

**Comment by mbleigh** [comment]
@tinaliang can you try it again? Tried to maybe fix the quoting issue.

**Pull request merged**